### PR TITLE
feat: Lookup unique address on all chains

### PR DESCRIPTION
**Pull request by Tritium-VLK**
Found myself writing a script to do this and thought it made the most sense in bal_addresses.

### CODE REVIEW

**Comment by Tritium-VLK** [comment]
Will add I wasn't quite sure where to put this.  It could also be a function in the AddrBook class, that can run without init.   This is a bit funny in one sense, but groups the address related function under the AddrBook class.

**Review by jalbrekt85 — COMMENTED** [review]
feel like having this function be a static method of AddrBook would be better. might be more intuitive to find if its organized inside the AddrBook with the rest of the methods

File: bal_addresses/addresses.py [hunk 1]
```diff
@@ -29,6 +29,25 @@
 ZERO_ADDRESS = "0x0000000000000000000000000000000000000000"
 
 
+def get_address_all_chains(search_string: str) -> dict:
```
Comment (jalbrekt85) [inline bal_addresses/addresses.py:32]:
```suggestion
def get_address_all_chains(search_string: str) -> dict[str, dict[str, str]]::
```

**Comment by Tritium-VLK** [comment]
> feel like having this function be a static method of AddrBook would be better. might be more intuitive to find if its organized inside the AddrBook with the rest of the methods

Care to propose a commit for how you think it best fits?  I agree it would maybe fit better as a method in the class, but the class is kind of cast for the most part by chain.

**Review by gosuto-inzasheru — COMMENTED** [review]
i agree moving it into the `AddrBook` itself would be cleaner.

yes `AddrBook` is chain aware, but it could just be reinit to a new chain in the for loop

**Comment by jalbrekt85** [comment]
> > feel like having this function be a static method of AddrBook would be better. might be more intuitive to find if its organized inside the AddrBook with the rest of the methods
> 
> Care to propose a commit for how you think it best fits? I agree it would maybe fit better as a method in the class, but the class is kind of cast for the most part by chain.

moved in [cc2284e](https://github.com/BalancerMaxis/bal_addresses/pull/448/commits/cc2284e133588fa968408bf613368548325835ee). 
you can access static methods without initializing the class. can just do: `AddrBook.get_address_all_chains()`

**Review by gosuto-inzasheru — APPROVED** [review]

**Pull request merged**